src/statedb/database_map.cpp

```diff
@@ -0,0 +1,92 @@
+#include "database_map.hpp"
+#include "utils.hpp"
+#include "scalar.hpp"
+
+void DatabaseMap::add(const string key, vector<Goldilocks::Element> value)
+{
+    lock_guard<recursive_mutex> guard(mlock);
+
+    mtDB[key] = value;
+    if (callbackOnChange) onChangeCallback();
+}
+
+void DatabaseMap::add(const string key, vector<uint8_t> value)
+{
+    lock_guard<recursive_mutex> guard(mlock);
+
+    programDB[key] = value;
+    if (callbackOnChange) onChangeCallback();
+}
+
+void DatabaseMap::add(MTMap &db)
+{
+    lock_guard<recursive_mutex> guard(mlock);
+
+    mtDB.insert(db.begin(), db.end());
+    if (callbackOnChange) onChangeCallback();
+}
+
+void DatabaseMap::add(ProgramMap &db)
+{
+    lock_guard<recursive_mutex> guard(mlock);
+
+    programDB.insert(db.begin(), db.end());
+    if (callbackOnChange) onChangeCallback();
+}
+
+bool DatabaseMap::findMT(const string key, vector<Goldilocks::Element> &value)
+{
+    lock_guard<recursive_mutex> guard(mlock);
+
```

Comment:
Perhaps, was better store on iterator position on variable, and after take this value as:
auto it = mtDB.find(key);
if (it != mtDB.end()) {
value = *it;
return true;
}